### PR TITLE
Update dependencies

### PR DESCRIPTION
**Pull request by trustin**
- Brave 5.12.3 -> 5.12.4
- Dropwizard 2.0.11 -> 2.0.12
- gRPC 1.30.2 -> 1.31.0
- Jetty 9.4.30 -> 9.4.31
- Micrometer 1.5.2 -> 1.5.3, 1.3.10 -> 1.3.11
- Reactor 3.3.7 -> 3.3.8
- Spring Boot 2.3.1 -> 2.3.2
- Shaded:
  - fastutil 8.3.1 -> 8.4.0
- Build:
  - Checkstyle 8.34 -> 8.35
  - Finagle ServerSets 20.6.0 -> 20.7.0
  - Mockito 3.4.4 -> 3.4.6

### CODE REVIEW

**Comment by trustin** [comment]
Do not review yet - I need to fix some compilation errors, remove deprecated usages and update frontend deps.

File: dependencies.yml [hunk 1]
```diff
@@ -164,14 +164,14 @@ com.typesafe.akka:
...
 
 gradle.plugin.net.davidecavestro:
   gradle-jxr-plugin: { version: '0.2.1' }
 
 io.dropwizard:
   dropwizard-core:
-    version: &DROPWIZARD_VERSION '2.0.11'
+    version: &DROPWIZARD_VERSION '2.0.12'
     javadocs:
     - https://www.javadoc.io/doc/io.dropwizard/dropwizard-core/2.0.11/
```
Comment (ikhoon) [inline dependencies.yml:176]:
```suggestion
    - https://www.javadoc.io/doc/io.dropwizard/dropwizard-core/2.0.12/
```

File: it/spring/boot2-tomcat8/build.gradle [hunk 2]
```diff
@@ -1,5 +1,5 @@
 final def SPRING_BOOT_VERSION = '2.1.15.RELEASE'
```
Comment (ikhoon) [inline it/spring/boot2-tomcat8/build.gradle:1]: Don't we need to upgrade to 2.1.16? 
https://github.com/spring-projects/spring-boot/releases/tag/v2.1.16.RELEASE

File: dependencies.yml (hunk 1 above)
Comment (trustin) [reply dependencies.yml:176]: Oh, thanks!

File: it/spring/boot2-tomcat8/build.gradle (hunk 2 above)
Comment (trustin) [reply it/spring/boot2-tomcat8/build.gradle]: We do. :sweat_smile:

**Comment by trustin** [comment]
Ready for reviews

File: spring/boot1-autoconfigure/build.gradle [hunk 3]
```diff
@@ -76,3 +82,9 @@ tasks.compileTestJava.dependsOn(project(':spring:boot2-autoconfigure').tasks.com
 tasks.withType(Checkstyle) {
     onlyIf { false }
 }
+
+// Graceful shutdown is supported only on Spring Boot 2
+// because Spring Boot 1 cannot convert a String to a Duration.
+tasks.withType(Test) {
+    exclude '**/ArmeriaGracefulShutdownConfigurationTest*'
+}
```
Comment (trustin) [inline spring/boot1-autoconfigure/build.gradle:90]: @heowc You might be interested in this :wink:

Comment (heowc) [reply spring/boot1-autoconfigure/build.gradle:90]: Oh... I'll check 😱

Comment (heowc) [reply spring/boot1-autoconfigure/build.gradle:90]: I think this is a side effect due to a problem from before it was created.
The current Spring-related dependency is looking at 2.x, not 1.5.x. I don't know when this has been done, but it has been going on before and the tests I've written passed normally. 

This means that with the addition of [these lines](https://github.com/line/armeria/pull/2945/files#diff-53dbe37e6482ebcaaa24ae04f6b9e1abR35-R40), the dependencies have returned to normal and the problem has only come to light. I think I should definitely fix this. WDYT?

Comment (trustin) [reply spring/boot1-autoconfigure/build.gradle:90]: Yep, that's correct. We can fix this by providing string-to-duration converter for older spring boot.

Comment (heowc) [reply spring/boot1-autoconfigure/build.gradle:90]: I will make a separate PR and fix it. 😭

**Pull request merged**